### PR TITLE
New source blackhole that never succeeds in authorizations

### PR DESCRIPTION
**Pull request by jrouzierinverse**

### CODE REVIEW

**Comment by extrafu** [comment]
What would be the purpose of that?

File: lib/pf/Authentication/Source/BlackholeSource.pm [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+  my $super_attributes = $self->SUPER::available_attributes;
+  my $own_attributes = [{ value => "username", type => $Conditions::SUBSTRING }];
+
+  return [@$super_attributes, @$own_attributes];
+}
+
+=head2 available_actions
+
+For an Null source, we limit the available actions to B<set role>, B<set access duration>, and B<set unreg date>.
```
Comment (extrafu) [inline lib/pf/Authentication/Source/BlackholeSource.pm]: Null?

File: lib/pf/Authentication/Source/BlackholeSource.pm [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+=cut
+
+has '+class' => (default => 'exclusive');
+
+has '+type' => (default => 'Blackhole');
+
+=head2 available_attributes
+
+Allow to make a condition on the user's email address.
```
Comment (extrafu) [inline lib/pf/Authentication/Source/BlackholeSource.pm:35]: Ugh?

File: lib/pf/Authentication/Source/BlackholeSource.pm (hunk 1 above)
Comment (jrouzierinverse) [inline lib/pf/Authentication/Source/BlackholeSource.pm]: This is a source that always fails it is the opposite of the Null source

Comment (extrafu) [inline lib/pf/Authentication/Source/BlackholeSource.pm]: Sure, but that's not the Null source, but the Blackhole one.

**Comment by extrafu** [comment]
Please merge it so we have that in v4.5.

**Pull request merged**